napi_js_props.md

```diff
@@ -0,0 +1,225 @@
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_has_named_property*
+
+#### Signature
+```
+napi_status napi_has_named_property(napi_env env,
+                                    napi_value object,
+                                    const char* utf8Name,
+                                    bool* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object to query
+- `[in]  utf8Name`: The name of the property whose existence to check
+- `[out] result`: Whether the property exists on the object or not
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_set_element*
+
+#### Signature
+```
+napi_status napi_set_element(napi_env env,
+                              napi_value object,
+                              uint32_t index,
+                              napi_value value);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object from which to set the properties
+- `[in]  index`: The index of the property to set
+- `[in]  value`: The property value
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_element*
+
+#### Signature
+```
+napi_status napi_get_element(napi_env env,
+                              napi_value object,
+                              uint32_t index,
+                              napi_value* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object from which to retrieve the property
+- `[in]  index`: The index of the property to get
+- `[out] result`: The value of the property
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_has_element*
+
+#### Signature
+```
+napi_status napi_has_element(napi_env env,
+                              napi_value object,
+                              uint32_t index,
+                              bool* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object to query
+- `[in]  index`: The index of the property whose existence to check
+- `[out] result`: Whether the property exists on the object or not
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Signature
+```
+napi_status napi_define_properties(napi_env env,
+                                   napi_value object,
+                                   size_t property_count,
+                                   const napi_property_descriptor* properties);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object from which to retrieve the properties
+- `[in]  property_count`: The number of elements in the `properties` array
+- `[in]  properties`: The array of property descriptors
+
+#### Description
+This method allows you to efficiently set multiple properties on a given object.
```

Comment:
There's a subtle difference between _setting_ properties and _defining_ properties.